hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/AbstractManagedType.java

```diff
@@ -161,56 +152,16 @@ public PersistentAttribute<? super J,?> getAttribute(String name) {
 	@Override
 	public PersistentAttribute<? super J,?> findAttribute(String name) {
 		// first look at declared attributes
-		PersistentAttribute<? super J,?> attribute = findDeclaredAttribute( name );
+		final PersistentAttribute<J,?> attribute = findDeclaredAttribute( name );
 		if ( attribute != null ) {
 			return attribute;
 		}
 
 		if ( getSuperType() != null ) {
-			attribute = getSuperType().findAttributeInSuperTypes( name );
-			if ( attribute != null ) {
-				return attribute;
-			}
-		}
-
-		for ( ManagedDomainType<?> subType : subTypes ) {
-			PersistentAttribute subTypeAttribute = subType.findSubTypesAttribute( name );
-			if ( subTypeAttribute != null ) {
-				if ( attribute != null && !isCompatible( attribute, subTypeAttribute ) ) {
-					throw new PathException(
-							String.format(
-									Locale.ROOT,
-									"Could not resolve attribute '%s' of '%s' due to the attribute being declared in multiple subtypes '%s' and '%s'",
-									name,
-									getTypeName(),
-									attribute.getDeclaringType().getTypeName(),
-									subTypeAttribute.getDeclaringType().getTypeName()
-							)
-					);
-				}
-				attribute = subTypeAttribute;
-			}
-		}
```

Comment:
I agree this code was probably wrong and we should *not* look for subtype attributes, but honestly I don't have any idea of the implications of this change, especially if HQL parsing relies on this method.

Reply:
This change is fine. I just moved the logic to a different, more sensible place. We *do* have tests for this stuff.